sklearn/model_selection/_validation.py

```diff
@@ -666,15 +689,11 @@ def _fit_and_score(
                 test_scores = error_score
                 if return_train_score:
                     train_scores = error_score
-            warnings.warn(
-                "Estimator fit failed. The score on this train-test"
-                " partition for these parameters will be set to %f. "
-                "Details: \n%s" % (error_score, format_exc()),
-                FitFailedWarning,
-            )
         result["fit_failed"] = True
+        result["fit_error"] = format_exc()
```

Comment:
Since `_fit_and_score` is private, I think we can remove `fit_failed` and use `fit_error` as a flag for when the fit failed. The only place that needs to be updated would be:

https://github.com/scikit-learn/scikit-learn/blob/2c2c6ca94b39bb5a2a5254681bb30389e8e989c7/sklearn/model_selection/_validation.py#L322